lib/btrfs_test.pm

```diff
@@ -40,13 +41,51 @@ way to get DBus-less environment is to enter rescue.target via systemctl.
 =cut
 sub snapper_nodbus_setup {
     my ($self) = @_;
+
+    # serial console prints text style as ascii chars in PS1 by default.
+    my $escseq = '[\]\[\(\)\e\d\w\s]*';
+
     if (script_run('! systemctl is-active dbus')) {
-        script_run('systemctl rescue', 0);
-        assert_screen('emergency-shell', 120);
-        type_password;
-        send_key 'ret';
-        $self->set_standard_prompt('root');
-        assert_screen 'root-console';
+        if (serial_terminal::select_virtio_console()) {
+            type_string "cp /usr/lib/systemd/system/rescue.service /usr/lib/systemd/system/.rescue.service.sav";
+            wait_serial "cp /usr/lib/systemd/system/rescue\\.service /usr/lib/systemd/system/\\.rescue\\.service\\.sav";
+            type_string "\n";
+            wait_serial("# ");
+
+            # Replace StandardInput in rescue.service file and add TTYPath.
+            type_string "sed -i 's@^StandardInput=.*\$\@StandardInput=tty\\nTTYPath='\$(tty)'\@g' \/usr\/lib\/systemd\/system\/rescue.service";
+            wait_serial 'sed -i \'s\@\^StandardInput=\.\*\$\@StandardInput=tty\\\\nTTYPath=\'\\$\(tty\)\'\@g\' \/usr\/lib\/systemd\/system\/rescue\.service';
+
+            type_string "\n";
+            wait_serial("# ");
```

Comment:
I'm not really sure, but can we use script_run for execution here instead of typing and waiting for serial? Or maybe create some wrapper which does these two calls and wait for the same string which was typed by default.